Backend/Controllers/MedicamentsController.cs

```diff
@@ -42,15 +40,33 @@ public async Task<ActionResult<GetMedicamentsDTO>> GetMedicaments()
             return Ok(new GetMedicamentsDTO(medicaments));
         }
 
-        //[HttpPost]
-        //public async Task<ActionResult> AddMedicament([FromBody] CreateMedicamentDTO dataFromBody)
-        //{
-        //    if (!IsValidApiRequest()) return InvalidHeaders();
+        [HttpPost]
+        public async Task<ActionResult> AddMedicament([FromBody] CreateMedicamentDTO dataFromBody)
+        {
+            ValidateCreateMedicamentDTO(dataFromBody);
 
-        //    var pharmaceuticalForm = Context.PharmaceuticalForms
-        //        .FirstOrDefaultAsync(p => (int) p.Id == dataFromBody.PharmaceuticalFormId);
+            var pharmaceuticalForm = Context.PharmaceuticalForms
+                .FirstOrDefaultAsync(p => (int)p.Id == dataFromBody.PharmaceuticalFormId);
 
+            Context.Medicaments.Add(new Medicament(dataFromBody, await pharmaceuticalForm));
+            await Context.SaveChangesAsync();
 
-        //}
+            return Created();
+        }
+
+        [AssertionMethod]
+        private static void ValidateCreateMedicamentDTO(CreateMedicamentDTO dto)
+        {
+            if (string.IsNullOrEmpty(dto.Name)) throw new ArgumentException("Name is empty!");
+            if (string.IsNullOrEmpty(dto.ActiveSubstance)) throw new ArgumentException("Active substance is empty!");
+            if (string.IsNullOrEmpty(dto.BarCode)) throw new ArgumentException("Bar code is empty!");
+            if (!dto.IsPrescriptionRequired.HasValue) throw new ArgumentException("Is prescription required is empty!");
+            if (!dto.IsReimbursed.HasValue) throw new ArgumentException("Is prescription required is empty!");
```

Comment:
Incorrect exception message

Reply:
How should it be changed?

Reply:
Is reimbursed required is empty or something, right now it's the same as the one above

Reply:
Fixed